bezier-rs/lib/src/lib.rs

```diff
@@ -331,6 +331,67 @@ impl Bezier {
 		};
 		bezier_starting_at_t1.split(adjusted_t2)[t2_split_side]
 	}
+
+	/// Find the roots of the linear equation ax + b
+	fn solve_linear(a: f64, b: f64) -> Vec<f64> {
+		let mut roots = Vec::new();
+		if b != 0. {
+			roots.push(-b / a);
+		}
+		roots
+	}
+
+	/// Find the roots of the linear equation ax^2 + bx + c
+	fn solve_quadratic(discriminant: f64, two_times_a: f64, b: f64, c: f64) -> Vec<f64> {
+		let mut roots = Vec::new();
+		if two_times_a != 0. {
+			if discriminant > 0. {
+				let root_discriminant = discriminant.sqrt();
+				roots.push((-b + root_discriminant) / (two_times_a));
+				roots.push((-b - root_discriminant) / (two_times_a));
+			} else if discriminant == 0. {
+				roots.push(-b / (two_times_a));
+			}
+		} else {
+			roots = Bezier::solve_linear(b, c);
+		}
+		roots
+	}
+
+	/// Returns two lists of `t`-values representing the local extrema of the `x` and `y` parametric curves respectively
+	/// The local extrema are defined to be points at which the derivative of the curve is equal to zero
+	fn _local_extrema(&self) -> [Vec<f64>; 2] {
+		match self.handles {
+			BezierHandles::Quadratic { handle } => {
+				let a = handle - self.start;
+				let b = self.end - handle;
+				let b_minus_a = b - a;
+				[Bezier::solve_linear(b_minus_a.x, a.x), Bezier::solve_linear(b_minus_a.y, a.y)]
```

Comment:
Not an actual request, but I'm wondering if there's a way to iterate over the `x` and `y` parameters in the `DVec2` rather than needing to write them out like this. It might be possible since you can also refer to `.x` and `.y` using indices. It's not that big of a deal though.

I thought it might be helpful for extending 3D Beziers later on, but we'd need a new struct for that anyways so it doesn't really matter.